src/server/qgsserverfilter.h

```diff
@@ -77,11 +83,50 @@ class SERVER_EXPORT QgsServerFilter
      * getFeature requests, sendResponse() might have been called several times
      * before the response is complete: in this particular case, sendResponse()
      * is called once for each feature before hitting responseComplete()
+     *
+     * This method is considered as deprecated and \see onSendResponse should
+     * be preferred.
      */
     virtual void sendResponse();
 
-  private:
+    /**
+     * Method called when the QgsRequestHandler is ready and populated with
+     * parameters, just before entering the main switch for core services.
+     *
+     * \return true if the call must propagate to the subsequent filters, false otherwise
+     *
+     * \since QGIS 3.22
+     */
+    virtual bool onRequestReady();
+
+    /**
+     * Method called when the QgsRequestHandler processing has done and
+     * the response is ready, just after the main switch for core services
+     * and before final sending response to FCGI stdout.
```

Comment:
is this specific to FCGI? Maybe better remove the mention to it.

Reply:
No, it's not specific to FCGI. Removed.